### PR TITLE
Upgrade to PHP 7.4 and resolve some PHPCS warnings.

### LINKED ISSUE #3975: Upgrade uiowa/01/02 application to PHP 7.4
- Update the box/config.yml PHP version to 7.4 and run `vagrant reload --provision`
  - https://github.com/uiowa/uiowa/pull/1445/commits/5a724e0431b1123da576e15f7b063a1d605af70f#diff-635c39d42eab7b5c0397077d62ead2a6e58f3cb52874b373710df96a85674831R958
- Sync some high-profile sites with `blt dsa`.
- Test out basic functionality and make any necessary code changes.
  - https://github.com/uiowa/uiowa/pull/1445/commits/82a07a0154345f2695a3905f3af8984316c8ec61
  - https://github.com/uiowa/uiowa/pull/1445/commits/c12f0e2d624f363da2b85692693c439702a34b6b
  - Not sure if an open-ended `composer update` is necessary or not. (https://github.com/uiowa/uiowa/pull/1445/commits/a8282906fd5bd4e3929fc7ecf6565e79e023fe37)
- Try running compat check as mentioned in #3675 if it isn't too much work.
- Merge feature branch into develop and deploy
- Upgrade dev environments to PHP 7.4. Test out:
  - authentication
  - caching
  - TBD.
- ??
- Update stage/prod and deploy code changes at the same time. 
- Update your local host PHP to 7.4. This can be done through various means depending on your setup. Most people are probably using Homebrew: `brew install php@7.4` then follow the instructions.

### PR DESCRIPTION
**Pull request by richardbporter**
Resolves #3975 

Currently on uiowa/01/02 dev which has been switched to PHP 7.4. Testing could happen in dev but if you want to do it locally, be sure to run `vagrant reload --provision` on this branch.

### Testing
- Authentication
- Caching
- Basic page content creation
- What else?

PHPCS warnings were fixed for the following which shouldn't result in functional changes but might be worth a quick test.
- brand.dev.drupal.uiowa.edu
  - Drush command
  - Lockup controller 
- SiteNow People person type filtering
-  SiteNow migrate post row event save
- uiowa_area_of_study block
- admissions_core
  - SectionComponentSubscriber (country_manager service)
  - ScholarshipsJumpMenu
  - AdmissionsCounselorsMapBlock
  - ScholarshipsBlock
  - ScholarshipsJumpMenuBlock

### CODE REVIEW

**Comment by richardbporter** [comment]
Since the PHP compat warnings are mixed in with the regular PHPCS warnings, I'm joeing rogue and doing a massive code cleanup as part of this.

**Comment by joewhitsitt** [comment]
pulling to test

**Comment by joewhitsitt** [comment]
I've tested a v2 default/custom site, v3 default/custom site touching on the following:

- syncs worked
- uli as user 1 works
- caching works as it relates to parent/child headings
- basic page creation (hiding sidebars, title, setting featured image, moderation, layout builder section and block)
- brand digest
- brand lockup
- people block person type filtering
  - looking at a warning i got when editing an exist person though
- admissions blocks seem to still be functioning
  - noticed counselors map is always yellow despite there not being a counselor for nebraska, etc. (currently an issue in PROD). going to take a quick peek.

**Comment by richardbporter** [comment]
@cory-skeers Do you know if this method is actually used?

https://github.com/uiowa/uiowa/blob/8933d66f3cd46ca04963f61aabc5f5960abb0ee1/docroot/modules/custom/sitenow_migrate/src/Plugin/migrate/source/Pages.php#L59

PHPCS flagged it as being unused. I would think since that class uses the `LinkReplaceTrait` which has an identical method, it would be unused. Not sure if the replacement was intentional or not.

https://github.com/uiowa/uiowa/blob/8933d66f3cd46ca04963f61aabc5f5960abb0ee1/docroot/modules/custom/sitenow_migrate/src/Plugin/migrate/source/LinkReplaceTrait.php#L225

**Comment by cory-skeers** [comment]
> @cory-skeers Do you know if this method is actually used?
> 
> https://github.com/uiowa/uiowa/blob/8933d66f3cd46ca04963f61aabc5f5960abb0ee1/docroot/modules/custom/sitenow_migrate/src/Plugin/migrate/source/Pages.php#L59
> 
> PHPCS flagged it as being unused. I would think since that class uses the `LinkReplaceTrait` which has an identical method, it would be unused. Not sure if the replacement was intentional or not.
> 
> https://github.com/uiowa/uiowa/blob/8933d66f3cd46ca04963f61aabc5f5960abb0ee1/docroot/modules/custom/sitenow_migrate/src/Plugin/migrate/source/LinkReplaceTrait.php#L225

I think it's a case of partial cleanup, where `postLinkReplace` got moved into the trait, but the default `sitenow_migrate` methodology of `manualLookup` that uses the migration maps and specified ordering never got moved over.

It can probably be removed. We haven't been running non-custom migrations, and there's probably a better way to get migration definitions dynamically that could apply to both default and custom migrations for link updating, and we'll still have the git record for reference.

**Comment by richardbporter** [comment]
Thanks! The only two PHPCS warnings left are related to to to-do items.

File: docroot/modules/custom/uiowa_area_of_study/uiowa_area_of_study.install [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+/**
+ * @file
+ * Install tasks.
+ */
+
+/**
+ * Clear all caches after block constructor change.
+ */
+function uiowa_area_of_study_update() {
```
Comment (richardbporter) [inline docroot/modules/custom/uiowa_area_of_study/uiowa_area_of_study.install]: @joewhitsitt caught this incorrectly named hook. Needs updating.

**Review by joewhitsitt — APPROVED** [review]
Wasn't able to get 7.4 local but tested most of these and other items in locally. Poked around in DEV.

**Comment by richardbporter** [comment]
> Wasn't able to get 7.4 local but tested most of these and other items in locally. Poked around in DEV.

I'd like to make sure others can get that working before merging this.

**Comment by richardbporter** [comment]
We seem to have lost https://github.com/uiowa/uiowa/pull/3983/commits/676b5ddeed9096f4b008ca44bb7696423303356a

**Comment by richardbporter** [comment]
🤦🏻 https://github.com/uiowa/uiowa/pull/3983/commits/3725d6fdee60dbce3b284dd3e8912f39ed33e8c5

**Pull request merged**